sql-plugin/src/main/scala/com/nvidia/spark/rapids/RapidsGdsStore.scala

```diff
@@ -120,14 +125,20 @@ class RapidsGdsStore(
       other.getSpillPriority, other.spillCallback)
   }
 
-  class BatchSpiller() {
+  class BatchSpiller() extends AutoCloseable {
     private val blockSize = 4096
     private[this] val spilledBuffers = new ConcurrentHashMap[File, Set[RapidsBufferId]]
     private[this] val pendingBuffers = ArrayBuffer.empty[RapidsGdsBatchedBuffer]
     private[this] val batchWriteBuffer = CuFileBuffer.allocate(batchWriteBufferSize, true)
     private[this] var currentFile = TempSpillBufferId().getDiskPath(diskBlockManager)
     private[this] var currentOffset = 0L
 
+    override def close(): Unit = {
+      pendingBuffers.foreach(_.free())
```

Comment:
I am no expert here. `spill` has a synchronized around it. Should we synchronize `close`? and maybe throw if we try to spill from a closed batch spiller?

Reply:
In general we don't synchronize on closing buffer stores.  If you're trying to close a store while it's actively trying to spill then there's larger-scale issues that would need to be solved that adding `synchronize` will not solve.  It will only serialize the contention between `spill` and `close` but not guarantee that `spill` does not get invoked after `close`.

If we need to deal with trying to spill on a closed store IMO that's a larger scale issue that needs to be addressed in all the stores, not just this one.